tests/Unit/DataStructures/Test_Variables.cpp

```diff
@@ -229,30 +229,51 @@ void test_variables_construction_and_access() noexcept {
       == "Variables(vector,scalar,scalar2)");
 }
 
-SPECTRE_TEST_CASE("Unit.DataStructures.Variables.Move",
-                  "[DataStructures][Unit]") {
-  Variables<tmpl::list<VariablesTestTags_detail::vector>> x(1, -2.0),
-      z(2, -3.0);
-  CHECK(&get<VariablesTestTags_detail::vector>(z)[0][0] == z.data());
-  Variables<tmpl::list<VariablesTestTags_detail::vector>> y = std::move(x);
-  x = std::move(z);
-  CHECK(
-      (x == Variables<tmpl::list<VariablesTestTags_detail::vector>>{2, -3.0}));
-  CHECK(&get<VariablesTestTags_detail::vector>(x)[0][0] == x.data());
+template <typename VectorType>
+void test_variables_move() noexcept {
+  MAKE_GENERATOR(gen);
+  UniformCustomDistribution<
+      typename tt::get_fundamental_type_t<typename VectorType::value_type>>
+      dist{-100, 100};
+  UniformCustomDistribution<size_t> sdist{2, 20};
+
+  size_t number_of_grid_points1 = sdist(gen),
+         number_of_grid_points2 = sdist(gen);
+  std::vector<typename VectorType::value_type> initial_fill_values{2};
+  fill_with_random_values(make_not_null(&initial_fill_values),
+                          make_not_null(&gen), make_not_null(&dist));
+
+  Variables<tmpl::list<VariablesTestTags_detail::vector_tag<VectorType>>>
+      target{number_of_grid_points1, initial_fill_values[0]};
```

Comment:
We normally use `target` to mean the `target points of a projection operation`. I'm not sure what it's supposed to mean in this context.